src/adapters/groq/chat.ts

```diff
@@ -145,6 +148,10 @@ export class GroqChatLLM extends ChatLLM<ChatGroqOutput> {
     };
   }
 
+  async embed(_input: BaseMessage[][], _options?: EmbeddingOptions): Promise<EmbeddingOutput> {
+    throw new NotImplementedError();
```

Comment:
Groq should have an embedding endpoint.

Reply:
We should merge this with NotImplemented where appropriate and I can create follow up issues to provide concrete implementations. I can use the ollama and watsonx embedding support for granite right now.

Reply:
i think we should get @michael-desmond unblocked here and then revisit providing the implementation for the remaining adapters

Reply:
We are realising daily; therefore, it would not be very mature to merge something that is not working. 
In addition, by implementing various adapters, you will discover whether the abstraction is correct. 

If you want to let @michael-desmond implement it, let's merge these changes into a pre-merge branch, and @michael-desmond will create a separate PR for that branch. Once everything is completed, we can merge it to the `main.` Not sooner, not later.

Reply:
Sure, but I would argue that not implemented is not equivalent to not working. Embedding for BAM, ollama and watsonx is working right now and is useful. I can go ahead and add groq embedding support but I have 2 days to develop a granite release tutorial which would benefit from the ollama embeddings functions.

Reply:
Based on the internal discussion, let's merge it without providing everyone with the implementations.